openedx/openedx-platform
# [ENT-760]:  Disable linking of personal accounts to enterprise customers via SSO

**Background/Problem to solve:**
Currently when coming in via SSO, the learner has option to create account or sign in with existing account. The linking account becomes a bit problematic from a flow perspective and customers are unhappy with how the data appears as there are a lot of non-company email addresses, etc.

**Request:**
Disable ability to link account when coming in via SSO. Force user to create an account with whatever credentials are being pass over.

**Considerations**
1. If user already has account on edX with an email address that matches the email address provided via SSO, then link this account and update the profile with any new information passed from company provider (ref:   ENT-759 DEPLOYED TO STAGE (BUSINESS)  ). At the end of this workflow there should still be a single edX learner account, now linked to the IdP and Enterprise Customer. Need to allow learner to reset password in this workflow.
2. Learner support may get requests to merge accounts. The statement will be until we have better messaging "Your company prefers you have a separate account connected to your learning associated with your work at the company. All work done prior to your company's partnership with edX is unfortunately not included in your company's program. We recommend you reach out to your manager about the best way to incorporate prior learning into your company's specific learning system" 

**Testing Instructions:**

1. Go to user admin page at https://business.sandbox.edx.org/admin/auth/user/
2. search for username `myself` (https://business.sandbox.edx.org/admin/auth/user/?q=myself) and delete the user that has this username if present.
3. Now go to admin page for third party auth at https://business.sandbox.edx.org/admin/third_party_auth/samlproviderconfig/ and click update for `Pied Piper` IdP.
4. Make suer `Sync learner profile data` is checked and `Skip registration form` is unchecked and hit save.
5. Now open account registration page https://business.sandbox.edx.org/register in a browser that does not have an active session.
6. Click on `Pied Piper` from `Create an account using` section
7. On testshib login page add credentials for `myself` user
8. Make sure you are redirected back to registration page of business sandbox and `email` and `Full Name`  fields are displayed and that `Full Name` field is grayed out and is not editable.
9. Add an email address and hit save and make sure account is created.
